doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,608 @@
+
+![State-Transition-Diagram](https://user-images.githubusercontent.com/61490193/223195784-05cc4e98-8883-4911-bca3-fded6a87d202.png)
+
+__*Figure 2: State Transition Diagram SwitchPort mode*__
+
+
+### Sequence Diagram for Adding Multiple VLANs
+ 
+
+![Add-Vlan](https://user-images.githubusercontent.com/61490193/223186173-b1eecf22-eef0-4ba1-86d1-8675a7b00f31.png)
+
+__*Figure 3: Sequence Diagram for adding Multiple VLANs*__
+
+**1.** The SONiC CLI will call the multiple vlan list parser which will parse multiple vlan(s) in a list and return.
+
+**2.** Check if the VLAN is in the range (2-4094) and whether the vid already exists or not.
+
+**3.** Check if vlan id already exist it will show an error message and truncate. Examples for such cases have been mentioned in the examples section below.
+
+**4.** After checking the vlan list in the range and vlan doesn’t already exist.
+
+**5.** CLI will add the VLAN from the list in the VLAN table.
+
+__*NOTE: This works same for Adding Multiple VLANs on a PortChannel*__
+
+
+### Seqence Diagram for Deleting Multiple VLANs
+
+
+![Delete-Vlan](https://user-images.githubusercontent.com/61490193/223189728-8545972f-c726-4d6a-a66c-984617e825c2.png)
+
+
+__*Figure 4: Sequence Diagram for deleting Multiple VLANs*__
+
+**1.** The SONiC CLI will call the multiple vlan list parser which will parse multiple vlan(s) in a list.
+
+**2.** Check if the VLAN is in the range .
+
+**3.** Check if vlan exists or not. If vlan does not exist it will show an error message and truncate. Examples for such cases have been mentioned in the examples section as well.
+
+**4.** Read the existing vlan table and delete vlans(s).
+
+
+__*NOTE: This works same for Deleting Multiple VLANs from a PortChannel*__
+
+
+### Sequence Diagram for Adding Multiple VLAN member on Trunk Port
+
+
+![Add-Vlan-Member](https://user-images.githubusercontent.com/61490193/223186389-94bb1c7f-0966-40a2-b569-38c0a56a5899.png)
+
+
+__*Figure 5: Sequence Diagram for Adding VLAN member on Trunk Port*__
+
+
+**1.**  The SONiC CLI will Call vlan_member_input_parser to add multiple vlan member(s).
+
+**2.**  Check for vlan_range.
+
+**3.**  Check port mode (access or trunk) for adding vlan members (untagged or tagged) on a port from the PORT table and return.
+
+**4.**  Add member vlan(s) in multiple commas or range list, all flag & except flag can be used as well.
+
+**5.**  Add the vlan member(s) in the vlan member table. 
+
+
+__*NOTE: This works same for Adding Multiple VLANs members on a PortChannel*__
+
+
+### Delete Multiple VLAN member on Trunk Port
+
+![Delete-Vlan-Member](https://user-images.githubusercontent.com/61490193/223199172-0ae75308-6e93-4176-9db6-3bcfbd02b948.png)
+
+
+__*Figure 6: Sequence Diagram for deleting VLAN member on Trunk Port*__
+
+**1.** The SONiC CLI will call the vlan_member_input_parser which will parse multiple vlan(s) in a list.
+
+**2.** Check if the members exist in the range .
```

Comment:
what happens if the requested vlan to be deleted is not part of the port's vlan membership? Do you reject the request?

Reply:
Yes, it will give an error saying the VLAN doesn't exist as a member of PORT and the command will abort.